raiden/tests/property/smart_contracts/test_nettingchannel.py

```diff
@@ -0,0 +1,333 @@
+            just(MINE),
+            integers(min_value=1, max_value=100),
+        )
+
+        return one_of(
+            deposit_op,
+            close_op,
+            update_transfer_op,
+            mine_op,
+        )
+
+    def execute_step(self, step):
+        op = step[0]
+
+        if op == DEPOSIT:
+            self.contract_deposit(step[1], step[2])
+
+        elif op == CLOSE:
+            self.contract_close(step[1], step[2])
+
+        elif op == UPDATE_TRANSFER:
+            self.contract_update_transfer(step[1], step[2])
+
+        elif op == MINE:
+            self.tester_state.mine(number_of_blocks=step[1])
+
+    def is_participant(self, pkey_address):
+        address = pkey_address[1]
+        return address in self.participant_addresses
+
+    def contract_deposit(self, deposit_amount, pkey_address):
+        token_balance = self.token.balanceOf(  # pylint: disable=no-member
+            pkey_address[1],
+            sender=pkey_address[0],
+        )
+
+        if not self.is_participant(pkey_address):
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit from non-participant didnt fail')
+
+        elif self.netting_channel.closed(sender=pkey_address[0]) != 0:  # pylint: disable=no-member
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit with closed channel didnt fail')
+
+        elif token_balance < deposit_amount:
+            try:
+                self.netting_channel.deposit(  # pylint: disable=no-member
+                    deposit_amount,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('deposit without balance didnt fail')
+
+        else:
+            self.netting_channel.deposit(  # pylint: disable=no-member
+                deposit_amount,
+                sender=pkey_address[0],
+            )
+
+    def contract_close(self, transfer, pkey_address):
+        transfer_data = transfer.encode()
+
+        if not self.is_participant(transfer.sender):
+            try:
+                self.netting_channel.close(  # pylint: disable=no-member
+                    transfer_data,
+                    sender=pkey_address[0],
+                )
+            except TransactionFailed:
+                pass
+            else:
+                raise ValueError('close with transfer data from a non participant didnt fail')
+
+        elif transfer.sender == pkey_address[1]:
```

Comment:
Same question as in the self-signed transfer for `updateTransfer`. Shouldn't both sender keys be the same for self-signed transfers?